lib/kino/smart_cell/chart_builder.ex

```diff
@@ -0,0 +1,180 @@
+defmodule Kino.SmartCell.ChartBuilder do
+  use Kino.JS, assets_path: "lib/assets/chart_builder"
+  use Kino.JS.Live
+  use Kino.SmartCell, name: "Chart builder"
+
+  @as_int ["width", "height"]
+  @as_atom ["data", "chart", "x_axis_type", "y_axis_type", "color_type"]
+
+  @impl true
+  def init(attrs, ctx) do
+    fields = %{
+      "chart" => attrs["chart"] || :bar,
+      "width" => attrs["width"],
+      "height" => attrs["height"],
+      "x_axis" => attrs["x_axis"],
+      "y_axis" => attrs["y_axis"],
+      "x_axis_type" => attrs["x_axis_type"],
+      "y_axis_type" => attrs["y_axis_type"],
+      "color" => attrs["color"],
+      "color_type" => attrs["color_type"]
+    }
+
+    {:ok, assign(ctx, fields: fields, options: %{}, vl_alias: nil, missing_dep: missing_dep())}
+  end
+
+  @impl true
+  def scan_binding(pid, binding, env) do
+    send(pid, {:scan_binding_result, binding, env})
```

Comment:
From the (unpublished) docs:

> **Important:** remember that data sent between processes is copied,
> so avoid sending large data structures. In particular, when looking
> at variables, instead of sending their values, extract and send
> only the relevant metadata.

So basically we want to move the work from `handle_info` here and then `send(pid, {:scan_binding_result, data_options, vl_alias})`, this way we send the minimal amount of data between processes :)